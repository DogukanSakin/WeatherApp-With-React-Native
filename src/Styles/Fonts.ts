const Fonts={
    lightFont:'OpenSans-Light',
    regularFont:'OpenSans-Regular',
    semiBoldFont:'OpenSans-SemiBold',
}
export default Fonts;