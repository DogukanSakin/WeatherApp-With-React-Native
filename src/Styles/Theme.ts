export const darkTheme={
    defaultColor:'#161616',
    greyColor:'#313030',
    darkerGreyColor:'#E3E3E3',
    componentColor:'white'
}
export const lightTheme={
    defaultColor:'white',
    greyColor:'#F5F4F4',
    darkerGreyColor:'#424040',
    componentColor:'#161616',
}