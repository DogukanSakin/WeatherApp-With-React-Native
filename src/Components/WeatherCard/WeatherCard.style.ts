import { StyleSheet } from 'react-native';
import Fonts from '../../Styles/Fonts';
import { darkTheme,lightTheme } from '../../Styles/Theme';

export const styles = (darkMode:boolean) =>StyleSheet.create({
    container:{
        flexDirection:'row',
        alignItems:'center',
        marginTop:7,
        marginBottom:7,
        margin:15,
        backgroundColor:darkMode ? darkTheme.greyColor : lightTheme.greyColor,
        borderRadius:50,
        paddingRight:10,
        paddingLeft:10,
        padding:10,  
    },
    cardIcon:{
        height:50,
        width:50
    },
    cityText:{
        textAlign:'center',
        flex:1,
        fontFamily:Fonts.lightFont,
        fontSize:25,
        color:darkMode?darkTheme.componentColor:lightTheme.componentColor
    },
    degreeText:{
        fontSize:25,
        fontFamily:Fonts.lightFont,
        color:darkMode?darkTheme.componentColor:lightTheme.componentColor
        
    },
})