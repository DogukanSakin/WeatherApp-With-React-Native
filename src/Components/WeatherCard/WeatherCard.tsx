import React,{FC} from 'react';
import { Text,View,Image } from 'react-native';
import {styles} from './WeatherCard.style';
interface ICardProps{
    darkMode?:boolean,
}
const WeatherCard : FC<ICardProps>=({darkMode=false})=>{
    const cardStyle=styles(darkMode);
    return(
        <View style={cardStyle.container}>
            <Image source={require('../../../assets/icons/Sunny.png')} style={cardStyle.cardIcon}></Image>
            <Text style={cardStyle.cityText}>Los Angeles</Text>
            <Text style={cardStyle.degreeText}>20Â°</Text>
        </View>
    )
}
export default WeatherCard;