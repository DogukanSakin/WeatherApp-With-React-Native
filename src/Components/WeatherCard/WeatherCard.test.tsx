import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import WeatherCard from './WeatherCard';
import { darkTheme,lightTheme } from '../../Styles/Theme';
const testCity={
    cityName:'TestCity',
    temp_c:20,
    conditionText:'Sunny'
}
test('should match with snapshot', () => { 
    const comp=render(<WeatherCard city={testCity} onTap={()=>{}}/>);
    expect(comp).toMatchSnapshot();
})
test('should be write all infos correctly', () => { 
    const comp=render(<WeatherCard city={testCity} onTap={()=>{}}/>);
    const cityText=comp.getByTestId('weather-card-cityText').children[0];
    expect(cityText).toBe(testCity.cityName);
    const degreeText=comp.getByTestId('weather-card-degreeText').children.join('');
    expect(degreeText).toContain(testCity.temp_c);
    const icon=comp.getByTestId('weather-card-icon');
    expect(icon.props.source['testUri']).toContain(testCity.conditionText); 
 })
 test('should be render default theme ', () => { 
    const comp=render(<WeatherCard city={testCity} onTap={()=>{}}/>);
    const card=comp.getByTestId('weather-card-container').props.style;
    expect(card['backgroundColor']).toBe(lightTheme.greyColor);
    
})
test('should be render dark theme if enable', () => { 
    const comp=render(<WeatherCard city={testCity} onTap={()=>{}} darkMode={true}/>);
    const card=comp.getByTestId('weather-card-container').props.style;
    expect(card['backgroundColor']).toBe(darkTheme.greyColor);
    
})

 test('should trigger onPress', () => { 
    const mockFunction=jest.fn();
    const comp=render(<WeatherCard city={testCity} onTap={mockFunction}/>);
    const buttonTocuhable=comp.getByTestId('weather-card-touchable');
    fireEvent(buttonTocuhable,'press');
    expect(mockFunction).toBeCalledTimes(1);
})