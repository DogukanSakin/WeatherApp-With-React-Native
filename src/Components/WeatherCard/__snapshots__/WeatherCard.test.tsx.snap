// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="weather-card-touchable"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F4F4",
        "borderRadius": 50,
        "flexDirection": "row",
        "margin": 15,
        "marginBottom": 7,
        "marginTop": 7,
        "padding": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
    testID="weather-card-container"
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/icons/Sunny.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
      testID="weather-card-icon"
    />
    <Text
      style={
        Object {
          "color": "#161616",
          "flex": 1,
          "fontFamily": "OpenSans-Light",
          "fontSize": 25,
          "textAlign": "center",
        }
      }
      testID="weather-card-cityText"
    >
      TestCity
    </Text>
    <Text
      style={
        Object {
          "color": "#161616",
          "fontFamily": "OpenSans-Light",
          "fontSize": 25,
        }
      }
      testID="weather-card-degreeText"
    >
      20
      Â°C
    </Text>
  </View>
</View>
`;
